src/mbedtls.c

```diff
@@ -688,8 +688,7 @@ _libssh2_dh_secret(_libssh2_dh_ctx *dhctx, _libssh2_bn *secret,
 void
 _libssh2_dh_dtor(_libssh2_dh_ctx *dhctx)
 {
-    mbedtls_mpi_free(*dhctx);
-    mbedtls_free(*dhctx);
+    _libssh2_bn_free(*dhctx);
```

Comment:
Unfortunately is the wrong call, it needs to be `_libssh2_mbedtls_bignum_free()` which calls `mbedtls_mpi_free()` as well as `mbedtls_free()`.

Reply:
Sorry, my bad.
The change was so easy that I did not try to compile it again. The funny thing is that in the commit comment, I specified the correct function... :)

Now it is correct and I have check that it compiles correctly.

Regards,
Adrian M.